File: validator.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

"""
validator.py - https://github.com/bacchilu/validate-visitor

An example of encapsulating a validation logic inside an object
(sort of Visitor Design Pattern)


Luca Bacchi <bacchilu@gmail.com> - http://www.lucabacchi.it
"""


class ValidatorException(Exception):

    pass


class Validator(object):

    def __init__(self, data):
        self.data = data
        self.errorMsg = None
        self.errorCode = 0

    def checkEmpty(self):
        if len(self.data) == 0:
            self.errorCode = 1
            self.errorMsg = u'Non pu\xc3\xb2 essere vuoto'
            raise ValidatorException()

    def checkSize(self):
        if len(self.data) < 3:
            self.errorCode = 2
            self.errorMsg = u'Troppo corto'
            raise ValidatorException()
        if len(self.data) > 10:
            self.errorCode = 3
            self.errorMsg = u'Troppo lungo'
            raise ValidatorException()

    def checkWhitespace(self):
        if len([c for c in self.data if c.isspace()]) > 0:
            self.errorCode = 4
            self.errorMsg = u'Niente spazi'
            raise ValidatorException()

    def validate(self):
        try:
            self.checkEmpty()
            self.checkSize()
            self.checkWhitespace()
            return True
        except ValidatorException:
            return False


if __name__ == '__main__':
    v = Validator('Cai a ')
    if not v.validate():
        print (u'%s (%d)' % (v.errorMsg, v.errorCode)).encode('utf-8')

File: validator.php
<?

class Validator {

    function Validator(data) {
        $this->data = data;
        $this->errorMsg = NULL;
        $this->errorCode = 0;
        $this->result = true;
    }

    function checkEmpty() {
        if ($this->result === false)
            return;

        if (strlen($this->data) === 0) {
            $this->errorCode = 1;
            $this->errorMsg = "Non può essere vuoto";
            $this->result = false;
            return;
        }
    }

    function checkSize() {
        if ($this->result === false)
            return;

        if (strlen($this->data) < 3) {
            $this->errorCode = 2;
            $this->errorMsg = "Troppo corto";
            $this->result = false;
            return;
        }
        if (strlen($this->data) > 10) {
            $this->errorCode = 3;
            $this->errorMsg = "Troppo lungo";
            $this->result = false;
            return;
        }
    }

    function checkWhitespace() {
        if ($this->result === false)
            return;

        if (preg_match("/\s/", $this->data)) {
            $this->errorCode = 4;
            $this->errorMsg = "Niente spazi";
            $this->result = false;
            return;
        }
    }

    function validate() {
        $this->checkEmpty();
        $this->checkSize();
        $this->checkWhitespace();
        return $this->result;
    }
}

?>